### PR TITLE
Feature/selection dialog search

### PR DESCRIPTION
**Pull request by bilgehankalkan**
Search implementation for selectionDialog. Can search in given `items`.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
`enableSearch` parameter will show search line above `recyclerView`. Also clear button visibility can set.

When dialog is expanded, corner radius will be removed slightly via `ObjectAnimator` and vice versa.

## How Has This Been Tested?
Tested on physical device with Android 7.1.

## Checklist:
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/Extensions.kt [hunk 1]
```diff
@@ -21,3 +31,59 @@ internal fun <T : ViewDataBinding> ViewGroup?.inflate(
...
+
+internal fun DialogFragment.animateCornerRadiusWithStateChanged() {
+    val expandAnimator =
+        ObjectAnimator.ofFloat(
+            binding.cardView,
+            "radius",
+            requireContext().pixel(R.dimen.dialogs_corner_radius),
+            0F
+        ).apply {
+            duration = 250
```
Comment (bilgehankalkan) [inline libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/Extensions.kt]: Couldn't get duration from resources, thats why its hardcoded.

File: libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/Extensions.kt [hunk 2]
```diff
@@ -21,3 +31,59 @@ internal fun <T : ViewDataBinding> ViewGroup?.inflate(
...
+        }
+    val collapseAnimator =
+        ObjectAnimator.ofFloat(
+            binding.cardView,
+            "radius",
+            0F,
+            requireContext().pixel(R.dimen.dialogs_corner_radius)
+        ).apply {
+            duration = 250
```
Comment (bilgehankalkan) [inline libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/Extensions.kt]: Couldn't get duration from resources, thats why its hardcoded.

File: libraries/dialogs/src/main/res/layout/fragment_dialog.xml [hunk 3]
```diff
@@ -78,34 +81,68 @@
...
                         app:layout_constraintTop_toBottomOf="@id/imageContent"
                         tools:text="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt" />
 
+                    <androidx.appcompat.widget.AppCompatEditText
+                        android:id="@+id/editTextSearch"
+                        style="@style/Trendyol.UIComponents.Dialogs.PrimaryText.Body"
+                        android:layout_width="0dp"
+                        android:layout_height="wrap_content"
+                        android:layout_marginBottom="@dimen/dialogs_margin_inner"
+                        android:background="#f5f5f5"
```
Comment (selimtoksal) [inline libraries/dialogs/src/main/res/layout/fragment_dialog.xml]: dimen or style

File: sample/src/main/res/layout/activity_dialogs.xml [hunk 4]
```diff
@@ -45,4 +45,14 @@
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toBottomOf="@id/button_agreement_dialog" />
+
+    <androidx.appcompat.widget.AppCompatButton
+        android:id="@+id/button_selection_with_search_dialog"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="32dp"
```
Comment (selimtoksal) [inline sample/src/main/res/layout/activity_dialogs.xml]: dimen

Comment (muratcanbur) [reply sample/src/main/res/layout/activity_dialogs.xml]: no worry about sample

File: libraries/dialogs/src/main/res/layout/fragment_dialog.xml [hunk 5]
```diff
@@ -25,8 +26,9 @@
                 android:layout_width="match_parent"
                 android:layout_height="?attr/actionBarSize"
                 android:background="#f5f5f5"
```
Comment (bsobe) [inline libraries/dialogs/src/main/res/layout/fragment_dialog.xml]: This color is used in multiple places. The hardcoded value may change with color resources.

Comment (bilgehankalkan) [reply libraries/dialogs/src/main/res/layout/fragment_dialog.xml]: Done.

File: libraries/dialogs/src/main/res/layout/fragment_dialog.xml (hunk 3 above)
Comment (bilgehankalkan) [reply libraries/dialogs/src/main/res/layout/fragment_dialog.xml]: Done.

**Pull request merged**